packages/smooth_app/lib/pages/offline_data_page.dart

```diff
@@ -0,0 +1,223 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:openfoodfacts/utils/ProductListQueryConfiguration.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/database/dao_product.dart';
+import 'package:smooth_app/database/dao_product_list.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/database/product_query.dart';
+import 'package:smooth_app/generic_lib/loading_dialog.dart';
+import 'package:smooth_app/pages/preferences/user_preferences_list_tile.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+class OfflineDataPage extends StatefulWidget {
+  const OfflineDataPage();
+  @override
+  State<OfflineDataPage> createState() => _OfflineDataPageState();
+}
+
+int _length = 0;
+double _size = 0;
+
+// TODO(ashaman999): update all the applocalizations
+class _OfflineDataPageState extends State<OfflineDataPage> {
+  Future<void> _refreshDetails(DaoProduct daoProduct) async {
+    _length = await daoProduct.getLength();
+    _size = await daoProduct.getSize();
+    setState(() {});
+  }
+
+  Future<String> _refreshLocalDatabase(DaoProduct daoproduct) async {
+    final List<String> barcodes = await daoproduct.getAllKeys();
+    if (barcodes.isEmpty) {
+      return 'List is Empty,Nothing to Refresh';
+    }
+    try {
+      final User user = ProductQuery.getUser();
+      // TODO(ashaman999): find a better way to do this
+      //Found that the max number of the barcodes i can query is 24
```

Comment:
Actually you're not limited to 24; there are parameters like pageNumber and pageSize. But anyway you'll need to download by chunks (bigger than 24 though), and that's a good opportunity to display a real progress (instead of a circular endless loop).
That would even be a good opportunity to code real background processes, but that's another story.

Reply:
I had also thought of background task here, once I start the work manager work I will change it here too